File: server/TraceFlow-SERVER-HTTPS/Code/Back/models/location/governorate.js
const { nanoid } = require('nanoid');
module.exports = (sequelize, DataTypes) => {
    return sequelize.define('Governorate', {
        governorateID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `gov_${nanoid(6)}`,
        },
        name: {
            type: DataTypes.STRING,
            unique: true
        },
        nameAr: {
            type: DataTypes.STRING
        },
        nameFr: {
            type: DataTypes.STRING
        },
        regionID: {
            type: DataTypes.STRING,
            allowNull: false,
            references: {
                model: 'Regions',
                key: 'regionID'
            }
        }
    });
};

File: Code/New folder/Mobile/lib/models/governorate.dart
class Governorate {
  final String governorateID;
  final String name;
  final String? nameAr;
  final String? nameFr;
  final String regionID;

  Governorate({
    required this.governorateID,
    required this.name,
    this.nameAr,
    this.nameFr,
    required this.regionID,
  });

  factory Governorate.fromJson(Map<String, dynamic> json) {
    return Governorate(
      governorateID: json['governorateID'] as String,
      name: json['name'] as String,
      nameAr: json['nameAr'] as String?,
      nameFr: json['nameFr'] as String?,
      regionID: json['regionID'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'governorateID': governorateID,
      'name': name,
      'nameAr': nameAr,
      'nameFr': nameFr,
      'regionID': regionID,
    };
  }
}